src/Symfony/Component/Yaml/Tests/InlineTest.php

```diff
@@ -420,4 +420,14 @@ public function testVeryLongQuotedStrings()
 
         $this->assertEquals($longStringWithQuotes, $arrayFromYaml['longStringWithQuotes']);
     }
+
+    public function testBooleanMappingKeysAreConvertedToStrings()
+    {
+        $this->assertSame(array('false' => 'foo'), Inline::parse('{false: foo}'));
```

Comment:
That should not be allowed actually, keys should be treated like any other value and evaluated, so boolean should not be allowed as a key.